packages/commonwealth/client/scripts/views/pages/Editor/DesktopEditorFooter.tsx

```diff
@@ -0,0 +1,65 @@
+import { Clipboard, DownloadSimple } from '@phosphor-icons/react';
+import React, { useCallback, useRef } from 'react';
+import './DesktopEditorFooter.scss';
+
+type DesktopEditorFooterProps = Readonly<{
+  onImportMarkdown?: (file: File) => void;
+  onSubmit?: () => void;
+}>;
+
+export const DesktopEditorFooter = (props: DesktopEditorFooterProps) => {
+  const { onImportMarkdown, onSubmit } = props;
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+
+  const fileHandler = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      if (event.target.files) {
+        console.log('nr files: ' + event.target.files.length);
+        onImportMarkdown?.(event.target.files[0]);
+      }
+    },
+    [onImportMarkdown],
+  );
+
+  const handleImportMarkdown = useCallback(() => {
+    if (fileInputRef.current) {
+      fileInputRef.current.click();
+    }
+  }, []);
```

Comment:
this seems unnecessary to store in memory

Reply:
I have to useCallback/useMemo in all this code to avoid the component from re-rendering because it will reset the cursor.